### PR TITLE
Doc strings

### PR DESCRIPTION
**Pull request by anwils2**
This PR:
- Adds docstring style definitions and parameter/return info for class methods.

Note: I didn't add anything for constructors specifically. I think I got everything else. Let me know what you would change or suggest changes directly!

### CODE REVIEW

File: Source/CustomVoice.cpp [hunk 1]
```diff
@@ -10,18 +10,31 @@
 
 #include "CustomVoice.h"
 
+// Indicates if this voice object is capable of playing the given sound.
+//
+// @param sound: A SynthesiserSound to be associated with this voice.
 bool CustomVoice::canPlaySound (juce::SynthesiserSound* sound)
```
Comment (robinjsu) [inline Source/CustomVoice.cpp:17]: do we want an `@return` here for consistency? Even though it's not really necessary for clarity...

Comment (anwils2) [reply Source/CustomVoice.cpp:17]: Good call, added [here](https://github.com/HudsWilSu/Subsynth/pull/15/commits/433d1bd34b6f538d47b91eda5355e5324dcc68e6)

File: Source/CustomVoice.cpp [hunk 2]
```diff
@@ -140,15 +178,22 @@ void CustomVoice::setFilter (int filterNum, float cutoff, float resonance)
     }
 }
 
+// Calls the setGainDecibels method on the gain data member. Self-explanitory.
```
Comment (robinjsu) [inline Source/CustomVoice.cpp]: nitofallnits:explanAtory

File: Source/PluginProcessor.cpp [hunk 3]
```diff
@@ -67,36 +72,47 @@ bool SubsynthAudioProcessor::isMidiEffect() const
...
 //==============================================================================
+
+// Called before playback starts, to let the processor prepare itself.
+//
+// @param sampleRate: Target sample rate
+// @param samplesPerBlock: A strong hint about the maximum number of samples 
+// that will be provided in each block.
 void SubsynthAudioProcessor::prepareToPlay (double sampleRate, int samplesPerBlock)
 {
     // Use this method as the place to do any pre-playback
```
Comment (robinjsu) [inline Source/PluginProcessor.cpp]: I vote we take this out since the docstring explains it

File: Source/PluginProcessor.cpp [hunk 4]
```diff
@@ -109,14 +125,20 @@ void SubsynthAudioProcessor::prepareToPlay (double sampleRate, int samplesPerBlo
     wfVisualiser.clear();
 }
 
+// Called after playback has stopped, to let the object free up any 
+// resources it no longer needs.
 void SubsynthAudioProcessor::releaseResources()
 {
     // When playback stops, you can use this as an opportunity to free up any
```
Comment (robinjsu) [inline Source/PluginProcessor.cpp]: Also can remove this comment

**Review by robinjsu — APPROVED** [review]
🚀

File: Source/CustomVoice.cpp [hunk 5]
```diff
@@ -87,13 +110,21 @@ void CustomVoice::prepareToPlay (double sampleRate, int samplesPerBlock, int num
...
+// @param params: A set of attack, decay sustain, release values in an ADSR::Parameters
+// object for the volume envelope to be set to.
 void CustomVoice::setADSR (juce::ADSR::Parameters parameters)
 {
     envelope.reset();
     envelope.setParameters (parameters);
 }
 
+// Changes which active oscillator between sine, square, saw, and triangle the
+// voice is using.
```
Comment (ahudson7) [inline Source/CustomVoice.cpp]: I would rephrase this to something like: "Changes the active oscillator the voice is using between sine, square, saw, and triangle.

File: Source/PluginProcessor.cpp [hunk 6]
```diff
@@ -216,6 +266,13 @@ void SubsynthAudioProcessor::changeVolume (double gain)
     }
 }
 
+// Calls the setFilter CustomVoice method to change the filter type, cutoff frequency,
+// and resonance of the state variable filter of each voice in the synth data member
+//
+// @param filterNum: An integer representation of the filter type to be set low pass,
+// band pass, high pass.
```
Comment (ahudson7) [inline Source/PluginProcessor.cpp]: Maybe rephrase to something like "An integer representation of the filter type to be set; options include low pass, band pass, and high pass."

Also, your phrasing choice may be a formatting thing I'm not aware of - if so just ignore me. (:

**Review by ahudson7 — APPROVED** [review]
Looks really great!  I added a couple stylistic comments, but you may be following conventions I'm not aware and if that's the case just ignore what I wrote.  Thanks for doing all of the writing for these!
